core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeExternalSorter.java

```diff
@@ -386,6 +386,18 @@ public void insertKVRecord(Object keyBase, long keyOffset, int keyLen,
   }
 
   /**
+   * Merges another UnsafeExternalSorters into this one, the other one will be emptied.
+   *
+   * @throws IOException
+   */
+  public void merge(UnsafeExternalSorter other) throws IOException {
+    other.spill();
+    spillWriters.addAll(other.spillWriters);
+    other.spillWriters.clear();
+    other.cleanupResources();
```

Comment:
Ordinarily, this would end up deleting the spill files, but it doesn't because of the `spillWriters.clear()` call above. If you end up updating this patch, mind adding a one-line comment to explain this (since it's a subtle point)?